src/app/public/modules/datepicker/datepicker-input.directive.ts

```diff
@@ -383,9 +388,11 @@ export class SkyDatepickerInputDirective
     if (value instanceof Date) {
       dateValue = value;
     } else if (typeof value === 'string') {
-      const date = this.dateFormatter.getDateFromString(value, this.dateFormat);
-      if (this.dateFormatter.dateIsValid(date)) {
-        dateValue = date;
+      if (!this.groupLogicService.hasBeenInitialized || this.groupLogicService.allGroupsHaveData(value)) {
```

Comment:
the logic for getting the date value lives with the datepicker input directive and that changes when a field masker is in play because it cannot update the value when any group has not been filled. I'm happy brainstorming but I think the "scope bleed" is inevitable unless you want to go back to having two separate directives